include/tvm/ir/expr.h

```diff
@@ -165,6 +166,20 @@ class RelayExprNode : public BaseExprNode {
   template <typename TTypeNode>
   inline const TTypeNode* type_as() const;
 
+  /*!
+   * \brief The storage execution scope (SEScope) for this node (the result of device planning).
+   *
+   * \note Unfortunately, the type of scope_ needs to be ObjectRef to avoid a circular import.
+   *       We can forward-declare the SEScope type for the getter function, but not for the field
+   * itself.
+   */
+  mutable ObjectRef scope_;
```

Comment:
I've been keeping the 'SE' part to avoid confusion with namespace scope. ie se_scope_.

Also that means we can globally rename to virtual_device_ which I think is the current front runner for preferred name.